components/brave_wallet_ui/common/async/lib.ts

```diff
@@ -286,3 +290,46 @@ export function refreshSitePermissions () {
     dispatch(WalletActions.setSitePermissions({ accounts: accountsWithPermission }))
   }
 }
+
+export async function signTrezorTransaction (apiProxy: APIProxyControllers, path: string, txInfo: TransactionInfo): Promise<SignHardwareTransactionType> {
+  const chainId = await apiProxy.ethJsonRpcController.getChainId()
+  const nonce = await apiProxy.ethTxController.getNonceForHardwareTransaction(txInfo.id)
+  if (!nonce.nonce) {
+    return { success: false, error: getLocale('braveWalletApproveTransactionError') }
+  }
+  txInfo.txData.baseData.nonce = nonce.nonce
```

Comment:
Is it necessary to mutate `txInfo` instead of creating a new one?

Reply:
this nonce is used by this transaction, I guess it is ok

Reply:
`txInfo` is  just a clone when getting it from the native side at that time.
If we fetch it again after nonce update, it will be the same nonce